### PR TITLE
Storm 1890 ensure we refetch static resources after package build

### PR DESCRIPTION
**Pull request by abellina**
Original pull request: https://github.com/apache/storm/pull/1486

I added in this PR the no-cache header for html pages in the / directory. To recap, this is to add cache-busting to Storm UI.

I also upgraded ring-core to 1.3.1 and compojure to 1.1.9 to match ring-core versions without changing too much. Versions above 1.1.9 seem to have issues with Clout not compiling for clojure 1.7. We could potentially go to a much newer version which fixes the issues (1.4.0), but that should be a different PR IMHO, and I am not sure how much we care since this is going away on conversion to Java.

### CODE REVIEW

**Comment by abellina** [comment]
@knusbaum @kishorvpatil @HeartSaVioR could you take a look at this PR? This will help with Storm UI caching issues in browsers. The build failures are unrelated.

**Comment by abellina** [comment]
ping

File: storm-core/src/clj/org/apache/storm/ui/core.clj [hunk 1]
```diff
@@ -18,7 +18,8 @@
   (:use compojure.core)
   (:use [clojure.java.shell :only [sh]])
   (:use ring.middleware.reload
-        ring.middleware.multipart-params)
+        ring.middleware.multipart-params
+        ring.middleware.multipart-params.temp-file)
```
Comment (knusbaum) [inline storm-core/src/clj/org/apache/storm/ui/core.clj]: Is this a necessary import?

Comment (knusbaum) [inline storm-core/src/clj/org/apache/storm/ui/core.clj]: Nevermind. Shading makes this import necessary. Multipart-params tries to load it at runtime and blows up if it's not already in the namespace.

**Comment by knusbaum** [comment]
+1

File: storm-core/src/clj/org/apache/storm/ui/helpers.clj [hunk 2]
```diff
@@ -36,22 +37,35 @@
...
   (fn [req]
     (.mark num-web-requests)
-    (handler req)))
+    (let [uri (:uri req)
+          res (handler req) 
+          content-type (response/get-header res "Content-Type")]
+      ;; check that the response is html and that the path is for a root page: a single / in the path 
+      ;; then we know we don't want it cached (e.g. /index.html)
+      (if (and (= content-type "text/html") 
+               (= 1 (StringUtils/countMatches uri "/")))
```
Comment (kishorvpatil) [inline storm-core/src/clj/org/apache/storm/ui/helpers.clj]: ```(count (re-seq #"/" url))``` could help find the # on "/" in more clojure like fashion.

**Comment by kishorvpatil** [comment]
One minor comment. The rest looks good.

**Comment by abellina** [comment]
@knusbaum, @kishorvpatil thanks for the review. @kishorvpatil I made the suggested change, thanks!

**Comment by abellina** [comment]
@HeartSaVioR, could you take a look at this PR?

**Comment by HeartSaVioR** [comment]
@abellina I don't have enough knowledge to front-end tech. But at a glance, I think concept is good.
@knusbaum @kishorvpatil Please go ahead reviewing & merging.

**Comment by abellina** [comment]
No problem. Thanks @HeartSaVioR.

**Comment by kishorvpatil** [comment]
Thank you @abellina. It looks good. +1.

**Pull request merged**